simulator/src/simulation_config.rs

```diff
@@ -0,0 +1,308 @@
+use chrono::{DateTime, Utc};
+use rand::{Rng, SeedableRng};
+use rand_distr::{Distribution, Exp, Normal};
+use serde::Deserialize;
+use std::ops::RangeInclusive;
+use supermusr_common::{Channel, DigitizerId, FrameNumber, Time};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) enum Expression {
+    FixedValue(f64),
+    FrameTransform(Transformation<f64>),
+}
+
+impl Expression {
+    fn value(&self, frame_index: usize) -> f64 {
+        match self {
+            Expression::FixedValue(v) => *v,
+            Expression::FrameTransform(frame_function) => {
+                frame_function.transform(frame_index as f64)
+            }
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "kebab-case", tag = "random-type")]
```

Comment:
Do the prefixes on the tag string really add anything or can they all just be "type"?

Reply:
I believe so, they make it easier to use the configuration file.